### PR TITLE
make clear resolution of ENV_VAR precedence over config property

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

**Comment by tamarrow** [comment]
fixes #632

File: dcos/config.py [hunk 1]
```diff
@@ -52,24 +54,36 @@ def get_config(mutable=False):
...
+    if config is None:
+        config = get_config()
+
+    section, subkey = split_key(name.upper())
+
+    env_var = None
+    if section == "CORE":
+        env_var = "DCOS_{}".format(subkey)
+        if subkey.startswith("DCOS") and os.environ.get(subkey):
+            env_var = subkey
```
Comment (jsancio) [inline dcos/config.py]: This is a bit confusing to read. Maybe we can write this as:

```
if subkey.startswith("DCOS") and os.environ.get(subkey):
    env_var = subkey
else:
    env_var = "DCOS_{}".format(subkey)
```

Let's write a short comment as to why we are doing this.

**Comment by jsancio** [comment]
LGTM @tamarrow . Do we need file an issue to print the help information around all the config properties and env variable. Maybe `dcos help config` or `dcos config describe`?

**Comment by tamarrow** [comment]
created issue #650

File: dcos/config.py (hunk 1 above)
Comment (tamarrow) [inline dcos/config.py]: updated

**Pull request merged**